baseapp/baseapp.go

```diff
@@ -161,6 +169,18 @@ func NewBaseApp(
 		option(app)
 	}
 
+	if app.mempool == nil {
+		app.SetMempool(mempool.NewSenderNonceMempool())
+	}
+
+	if app.prepareProposal == nil {
+		app.SetPrepareProposal(app.DefaultPrepareProposal())
+	}
+
+	if app.processProposal == nil {
+		app.SetProcessProposal(app.DefaultProcessProposal())
+	}
```

Comment:
Could you please remind me - in Osmosis, both of these are custom to perform a nonce validation check while relying on Tendermint's FIFO, is that correct?

Reply:
Correct!